runtime/compiler/control/JITServerHelpers.cpp

```diff
@@ -921,6 +921,16 @@ JITServerHelpers::postStreamFailure(OMRPortLibrary *portLibrary, TR::Compilation
       _waitTimeMs *= 2; // Exponential backoff
       }
    _nextConnectionRetryTime = current_time + _waitTimeMs;
+
+   if (_serverAvailable && TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerboseJITServerConns))
+      {
+      TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+                                     "t=%6u Lost connection to the server (serverUID=%llu)",
+                                     (uint32_t) compInfo->getPersistentInfo()->getElapsedTime(),
+                                     compInfo->getPersistentInfo()->getServerUID());
+      compInfo->getPersistentInfo()->setServerUID(0);
```

Comment:
Is it correct to reset the server UID here? What if only one compilation thread experiences a stream failure and the others still work fine?

Reply:
We set `_serverAvailable` to false here, so I am working under the assumption that if one stream fails, the rest will fail as well.